runbotics/runbotics-orchestrator-ui/src/src-app/views/process/EditProcessDialog.tsx

```diff
@@ -29,71 +33,147 @@ type EditProcessDialogProps = {
-                            variant="outlined"
-                        />
-                        <Box display="flex" justifyContent="space-between">
-                            <BotCollectionComponent
-                                selectedBotCollection={formState.botCollection}
-                                onSelectBotCollection={(botCollection) => setFormState({ ...formState, botCollection })}
-                            />
-                            <BotSystemComponent
-                                selectedBotSystem={formState.system}
-                                onSelectBotSystem={(system) => setFormState({ ...formState, system })}
+                    <EditProcessDialogTextFields
+                        processFormState={processFormState}
+                        setProcessFormState={setProcessFormState}
+                        formValidationState={formValidationState}
+                        setFormValidationState={setFormValidationState}
+                    />
+                    <Autocomplete
+                        fullWidth
+                        value={selectedTagsNames}
+                        renderInput={(params) => (
+                            <TextField
+                                {...params}
+                                label='Tags'
+                                onChange={handleTagSearch}
+                                onKeyDown={handleAddByEnter}
+                                placeholder={
+                                    selectedTagsNames.length
+                                        ? ''
+                                        : translate('Process.Edit.Form.Fields.Placeholder.TagSearch')
+                                }
                             />
-                        </Box>
-                        <FormControlLabel
-                            control={(
-                                <Switch
-                                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
-                                        setFormState({ ...formState, isPublic: e.target.checked });
-                                    }}
-                                    checked={formState.isPublic}
-                                />
-                            )}
-                            label={translate('Process.Edit.Form.Fields.Public.Label')}
-                            labelPlacement="start"
-                            sx={{
-                                width: 'fit-content', marginLeft: '0', padding: '0.5rem',
-                            }}
-                        />
-                    </FormControl>
+                        )}
+                        options={getSearchedTagNames()}
+                        onChange={(_, value) => handleTagChange(value)}
+                        getOptionDisabled={() => !(selectedTagsNames.length < MAX_NUMBER_OF_TAGS)}
+                        filterOptions={(options) => options}
```

Comment:
Can you briefly explain the purpose of this prop?

Reply:
:0